proxy/http/HttpSM.cc

```diff
@@ -2737,6 +2783,24 @@ HttpSM::tunnel_handler_post(int event, void *data)
     break;
   case HTTP_SM_POST_SUCCESS:
     // It's time to start reading the response
+    if (is_waiting_for_full_body) {
+      is_waiting_for_full_body  = false;
+      is_using_post_buffer      = true;
+      client_request_body_bytes = this->postbuf_buffer_avail();
+
+      APIHook *hook = api_hooks.get(TS_HTTP_REQUEST_BUFFER_READ_COMPLETE_HOOK);
+      if (hook) {
+        post_buffer_reader = this->_postbuf.postdata_copy_buffer->clone_reader(this->_postbuf.postdata_copy_buffer_start);
+        while (hook) {
+          hook->invoke(TS_EVENT_HTTP_REQUEST_BUFFER_COMPLETE, this);
```

Comment:
Why do not move plugin logic to `state_api_callout`.

Reply:
Good idea! Fixed.